relayer/crates/starknet-integration-tests/src/tests/ics20.rs

```diff
@@ -589,16 +595,308 @@ fn test_starknet_ics20_contract() -> Result<(), Error> {
         );
 
         let balance_starknet_b_step_2 = starknet_chain
-            .query_balance(address_starknet_b, &ics20_token_address)
+            .query_token_balance(&ics20_token_address, address_starknet_b)
             .await?;
 
         info!(
-            "starknet balance after transfer back: {:?}",
+            "starknet balance after transfer back: {}",
             balance_starknet_b_step_2
         );
 
         assert_eq!(balance_starknet_b_step_2.quantity, 0u64.into());
 
+        // send starknet native token to cosmos
+
+        let wallet_starknet_relayer = &starknet_chain_driver.relayer_wallet;
+        let address_starknet_relayer = &wallet_starknet_relayer.account_address;
+
+        let erc20_token_supply = 1_000_000_000u128;
+
+        let erc20_token_address = {
+            let deploy_message = DeployErc20TokenMessage {
+                name: "token".into(),
+                symbol: "token".into(),
+                fixed_supply: erc20_token_supply.into(),
+                recipient: *address_starknet_relayer,
+                owner: *address_starknet_relayer,
+            };
+
+            let calldata = StarknetCairoEncoding.encode(&deploy_message)?;
+
+            starknet_chain
+                .deploy_contract(&erc20_class_hash, false, &calldata)
+                .await?
```

Comment:
Can’t we use the [pre-deployed token contracts](https://0xspaceshard.github.io/starknet-devnet-rs/docs/balance) on devnet? That way, we wouldn’t need to deploy another token. This one though can be a good another testing scenario, but maybe not the primary use case.

Reply:
bf09dbe